rtc/encryption.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: default
+title: Encryption information for the RTC protocol
+title_nav: RTC Encryption help
+description: Useful information for understanding how encryption is used with RTC
+keywords: rtc encrypt encryption decrypt decryption key rotate signature
+---
+
+This documentation is in progress. Please contact us with any suggestions you think should be here.
+
+> **Caution**: The advice on this page may not guarantee a secure connection. If data secrecy is important to you please consult a security professional.
+
+## Encryption keys are mandatory
+
+RTC requires an encryption key so {{site.companyname}} can prove to ourselves (and any interested customers) that the RTC server does not store document content in a format that we are able to read. We are very focussed on privacy and end-to-end encryption for this feature.
+
+## The difference between content encryption and JWT signing
+
+Encryption is a two way process (encryption and decryption) that transforms data into an encrypted form. Modern encryption algorithms produce an output that is indistinguishable from random data without the decryption key. RTC performs symmetric encryption (meaning encryption and decryption use the same key) on all document content, but the key is never sent to the server. As the key only exists on the client endpoints, the term for this is end-to-end encryption.
+
+Signing is a one way process where the data is not modified and can still be read by anyone. A signature is transmitted along with the data and can be used to validate the data has not been modified since it was approved by whoever has the signing key. RTC requires an [asymmetric signing algorithm]({{site.baseurl}}/rtc/jwt-authentication/#supportedalgorithms) for JWT identity tokens. Asymmetric signing uses a public/private key pair: signatures are created with the private key and verified using the public key. The public key stored on the Tiny Cloud servers can only be used to verify signatures, not create new ones, so it is quite safe for us to hold or cache without security implications should it be disclosed.
+
+## Choosing an encryption key
+
+RTC encryption keys can be any unicode string. Our demo application for example uses the simple fixed key `"not a very secret 🔑"`. Even simple keys provide some obfuscation and protection if the key is disclosed; for details on this see [RTC encryption details](#rtcencryptiondetails) below. Simple fixed keys can be a good way to get a test environment up and running but we strongly recommend using secure keys when collaborating on private production data.
```

Comment:
- Can we change `unicode` to `utf-8` or appropriate throughout these docs?
- I am strongly against suggesting that people use insecure things, and then adding a "but please change it before you go to prod" because people never change it before they go to prod
- "provide some obfuscation and protection if the key is disclosed" sounds like one of the words should have a negative in it

Reply:
> I am strongly against suggesting that people use insecure things, and then adding a "but please change it before you go to prod" because people never change it before they go to prod

I generally agree, but in this case doing that will raise the barrier to entry too high. RTC is far more complicated to set up than any feature we've ever released before, so we need some way to ease people into it. We've even considered backing down on the signed JWT requirement for test environments but that's not viable on our cloud services.

Reply:
Unicode came from the code, and is technically more correct (JavaScript strings are unicode not utf-8). But if we use utf-8 in other TinyMCE docs I'm happy to change it.

I'll see if I can reword the obfuscation piece, it's already up to the third rewrite. Explaining this stuff in simple terms is _hard_.

Reply:
There are 3 references to `utf-8` in our docs; two in enterprise spelling and one in a config block. There are 8 uses of `unicode` (10 after including RTC).

For now, I'll stick to the more common term we use; perhaps consult Tyler about changing it across our entire documentation site.

I've pushed up a rewrite of this section; I even split off a sub heading "fixed keys are not secure". Hopefully that helps.